src/app/src/pages/eval/components/EvalOutputCell.tsx

```diff
@@ -153,10 +153,34 @@ function EvalOutputCell({
 
   const [lightboxOpen, setLightboxOpen] = React.useState(false);
   const [lightboxImage, setLightboxImage] = React.useState<string | null>(null);
-  const toggleLightbox = (url?: string) => {
-    setLightboxImage(url || null);
-    setLightboxOpen(!lightboxOpen);
-  };
+
+  // Memoized to maintain stable reference across renders, preventing
+  // unnecessary re-renders of markdown components that use this callback.
+  // Uses functional update to avoid stale closure issues.
+  // @see https://github.com/promptfoo/promptfoo/issues/969
+  const toggleLightbox = useCallback((url?: string) => {
+    setLightboxImage(url ?? null);
+    setLightboxOpen((prev) => !prev);
+  }, []);
+
+  // Memoized components object for ReactMarkdown to prevent re-renders.
+  // Creating this inline would cause ReactMarkdown to re-render on every
+  // parent render, even when content hasn't changed.
```

Comment:
instead of creating inline, could this be extracted into a helper component or something so react can handle it under the hood?